api/src/db/models/user_models.py

```diff
@@ -421,3 +424,16 @@ class AgencyUserRole(ApiSchemaTable, TimestampMixin):
 
     role_id: Mapped[uuid.UUID] = mapped_column(ForeignKey(Role.role_id), primary_key=True)
     role: Mapped[Role] = relationship(Role)
+
+
+class UserProfile(ApiSchemaTable, TimestampMixin):
+    __tablename__ = "user_profile"
+
+    user_id: Mapped[uuid.UUID] = mapped_column(
+        ForeignKey(User.user_id), primary_key=True, index=True
+    )
```

Comment:
A primary key column is indexed automatically by Postgres - can skip doing that.
```suggestion
    user_id: Mapped[uuid.UUID] = mapped_column(
        ForeignKey(User.user_id), primary_key=True
    )
```
Some tables might have an index on it when we have a group of columns serving as a primary key, although I'm not sure how much that helps.